### ISSUE TITLE
2.2 Dev: General Discussion

### ISSUE THREAD

**Issue by tajmone** [AnssiR66/AlanStdLib#70]
2.2 Dev: General Discussion

This Issue is a thread for the general discussion and planning for the upcoming Library v2.2 release. For discussing specific topics, see the dedicated issues:

* [2.2 Dev: Library Code Discussion]
* [2.2 Dev: Documentation Discussion]

Discussion threads are grouped under the [Release 2.2.0] milestone.

[Release 2.2.0]: https://github.com/AnssiR66/AlanStdLib/milestone/2
[2.2 Dev: Library Code Discussion]: https://github.com/AnssiR66/AlanStdLib/issues/68
[2.2 Dev: Documentation Discussion]: https://github.com/AnssiR66/AlanStdLib/issues/69

**Comment by tajmone** [comment]
# Keeping Topics in Separate Threads 

To avoid creating long threads that end up discussing off-topic issues, we should try to create an individual issue for each topic discussed, and mark the issue under the [Release 2.2.0] milestone.

As for the individual tasks, tracking issues will be created under the milestone for that topic:

* [2.2 Dev: Documentation]
* [2.2 Dev: Library Code]

This should make it easier to track the progress in an orderly manner, organized by topics. Most important, we can close issues as they are solved (whereas off topic discussion inside an issue force to keep the issue open even when its original topic is solved).

* [StdLib 2.2 Dev]  (_project_) — contains all the pending tasks for the 2.2 release, organized by status:
    - To Do
    - In Progress
    - Pending Approval
    - Done
* [Release 2.2.0]  (_milestone_) — groups the issues for discussing and planning the pending changes, organized by topic. Doesn't contain issues for the specific tasks tracking.


[StdLib 2.2 Dev]: https://github.com/AnssiR66/AlanStdLib/projects/2



[Release 2.2.0]: https://github.com/AnssiR66/AlanStdLib/milestone/2
[2.2 Dev: Documentation]: https://github.com/AnssiR66/AlanStdLib/milestone/6
[2.2 Dev: Library Code]: https://github.com/AnssiR66/AlanStdLib/milestone/5




[2.2 Dev: Library Code Discussion]: https://github.com/AnssiR66/AlanStdLib/issues/68
[2.2 Dev: Documentation Discussion]: https://github.com/AnssiR66/AlanStdLib/issues/69

**Comment by tajmone** [comment]
# Deciding on Pending Ideas/Proposals

Anssi, there are still a few open issues with ideas I had proposed to enhance the StdLib, based on my work on the Italian Library.
When you have time, could you look at the issues shown at this link:

* [List of ideas pending decision]  (automatically filtered)

Maybe the pre-2.2 release time could be a good occasion for you to decide on  which of these proposals should be approved and which discarded, so we can close the issues for the discarded ones, and mark the approved ones for impelementation (labels: "approved" and "pending implementation").



[List of ideas pending decision]: https://github.com/AnssiR66/AlanStdLib/issues?utf8=%E2%9C%93&q=is%3Aopen+label%3A%22%3Abulb%3A+idea%22+-label%3A%22%3Abulb%3A+approved+%3Aheavy_check_mark%3A%22+

**Comment by tajmone** [comment]
# Branches Clean Up

Anssi, I propose we do some cleanup regarding the branches, to avoid confusion.

We could delete the old clorhing work branches that were discarded in favour of `dev-clothing`:

- [ ] `clothing`
- [ ] `clothing-solution2`
- [ ] `try-dispose-worn`

I can keep a copy of these branches in my clone of this repository, just in case we'd like to look at them again. In any case, we can always keep local copies of them. But we should get rid of them on the repository for they might create confusion in the future.

I also suggest keeping the `dev-clothing` branch until the release is ready, but:

- [ ] rename `dev-clothing` to `dev-2.2.0-new-clothing-sys`

(if I do rename the branch then you need to update your local copy accordingly, and if you have branched from it it could be a problem).

Since this branch has been squashed we're keeping it just in case we need to untangle something by looking up the history changes. But if we adopt a consistent naming convention referencing the the release to which every dev branch refers to, then we could just keep them on the repository indefinitely.

**Comment by AnssiR66** [comment]
Yes, this is fine with me. We can simplify the branches this way.

**Comment by tajmone** [comment]
Done!

**Comment by tajmone** [comment]
# Closing This Issue

We no longer need to keep this thread alive now that [Discussions] is enabled, since we can create new dedicated threads in the [Library Improvements] and [Tech & Dev Stuff] categories for these topics.

[Library Improvements]: ./categories/library-improvements
[Tech & Dev Stuff]: ./categories/tech-dev-stuff
[Discussions]: https://github.com/AnssiR66/AlanStdLib/discussions
